frontend/scripts/runE2eWithServer.ts

```diff
@@ -0,0 +1,105 @@
+import tcpPortUsed from 'tcp-port-used'
+import path from 'path'
+import process from 'process'
+
+const PORT = {
+  FE: 4321,
+  BE: 4322,
+  STUB: 4323,
+}
+
+const DIR = {
+  FE: path.resolve(__dirname, '../'),
+  BE: path.resolve(__dirname, '../../backend'),
+  STUB: path.resolve(__dirname, '../../stubs'),
+}
+
+const main = async (args: string[]) => {
+  const { $ } = await import('execa')
+
+  const checkPorts = async () => {
+    const ports = Object.values(PORT)
+    const checkPorts = ports.map((port) => tcpPortUsed.check(port))
+
+    const usingPorts = await Promise.all(checkPorts).then((result) => ports.filter((_, idx) => result[idx]))
+    const availablePorts = ports.filter((port) => !usingPorts.includes(port))
+
+    console.log(`E2E: Currently available ports: ${availablePorts.join(',')}`)
+
+    return {
+      availablePorts,
+      usingPorts,
+    }
+  }
+
+  const startFE = () => $({ cwd: DIR.FE, stderr: 'inherit', shell: true })`pnpm run start`
+
+  const startBE = () =>
+    $({
+      cwd: DIR.BE,
+      stderr: 'inherit',
+      shell: true,
+    })`./gradlew bootRun --args='--spring.profiles.active=local --MOCK_SERVER_URL=http://localhost:4323'`
+
+  const startSTUB = () =>
+    $({
+      cwd: DIR.STUB,
+      stderr: 'inherit',
+      shell: true,
+    })`docker-compose up`
```

Comment:
should not use `multiple tech`
Maybe you can wrap all kinds of services into 1 `docker-compose.yaml` and then start it. ref to [this](https://stackoverflow.com/questions/53649873/docker-compose-mapping-local-directory-to-dockerfile-volume)